packages/edit-navigation/src/store/utils.js

```diff
@@ -172,3 +159,147 @@ export function computeCustomizedAttribute(
 		return omit( menuItemsByClientId[ block.clientId ] || {}, '_links' );
 	}
 }
+
+/**
+ * Convert block attributes to menu item fields.
+ *
+ * Note that nav_menu_item has defaults provided in Core so in the case of undefined Block attributes
+ * we need only include a subset of values in the knowledge that the defaults will be provided in Core.
+ *
+ * See: https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L438.
+ *
+ * @param {Object} blockAttributes the block attributes of the block to be converted into menu item fields.
+ * @param {string} blockAttributes.label the visual name of the block shown in the UI.
+ * @param {string} blockAttributes.url the URL for the link.
+ * @param {string} blockAttributes.description a link description.
+ * @param {string} blockAttributes.rel the XFN relationship expressed in the link of this menu item.
+ * @param {string} blockAttributes.className the custom CSS classname attributes for this block.
+ * @param {string} blockAttributes.title the HTML title attribute for the block's link.
+ * @param {string} blockAttributes.type the type of variation of the block used (eg: 'Post', 'Custom', 'Category'...etc).
+ * @param {number} blockAttributes.id the ID of the entity optionally associated with the block's link (eg: the Post ID).
+ * @param {string} blockAttributes.kind the family of objects originally represented, such as 'post-type' or 'taxonomy'.
+ * @param {boolean} blockAttributes.opensInNewTab whether or not the block's link should open in a new tab.
+ * @return {Object} the menu item (converted from block attributes).
+ */
+export const blockAttributesToMenuItem = ( {
+	label = '',
+	url = '',
+	description,
+	rel,
+	className,
+	title: blockTitleAttr,
+	type,
+	id,
+	kind,
+	opensInNewTab,
+} ) => {
+	// For historical reasons, the `core/navigation-link` variation type is `tag`
+	// whereas WP Core expects `post_tag` as the `object` type.
+	// To avoid writing a block migration we perform a conversion here.
+	// See also inverse equivalent in `menuItemToBlockAttributes`.
+	if ( type && type === 'tag' ) {
+		type = 'post_tag';
+	}
+
+	return {
+		title: label,
+		url,
+		...( description?.length && {
+			description,
+		} ),
+		...( rel?.length && {
+			xfn: rel?.trim().split( ' ' ),
+		} ),
+		...( className?.length && {
+			classes: className?.trim().split( ' ' ),
+		} ),
+		...( blockTitleAttr?.length && {
+			attr_title: blockTitleAttr,
+		} ),
+		...( type?.length && {
+			object: type,
+		} ),
+		...( kind?.length && {
+			type: kind?.replace( '-', '_' ),
+		} ),
+		// Only assign object_id if it's a entity type (ie: not "custom").
+		...( id &&
+			'custom' !== type && {
+				object_id: id,
+			} ),
+		...( opensInNewTab && {
+			target: NEW_TAB_TARGET_ATTRIBUTE,
```

Comment:
If opensInNewTab is false, target should be added as '' to allow the navigation editor to unset this value. This is because of how to how we're applying these values in https://github.com/WordPress/gutenberg/blob/7497fa6a524b4c366bea540b0dd619236dc89782/packages/edit-navigation/src/store/utils.js#L147-L149 

Menu editor ✅ 
Navigation Editor ❌ (Can't unset)
Post Editor ✅

I tested this in each editor with the following steps:

1. set opens in new tab
2. save
3. reload
4. verify opens in new tab
5. unset this value
6. save
7. reload




https://user-images.githubusercontent.com/1270189/116735837-eca60a00-a9a3-11eb-88f6-778bb2afaf89.mp4

Reply:
Great spot. Fixed and added specific test coverage.